### PR TITLE
Endpoint Security Without env variables

### PR DESCRIPTION
**Pull request by saquino0827**
### Information
- This is the endpoint security branch with environment variables removed. After talking with @clydet, we decided that getting the environment variables would be cumbersome and would like to add the environment variables as a separate task.

### Changes proposed in this PR:
- Add security on endpoints `/cpc/**` to restrict access unless you have a authenticated and authorized JWT token.

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [x] Added and updated JavaDocs for non-test classes and methods.
- [x] No local design debt. Do you feel that something is "ugly" after your changes?
- [x] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/security/JwtAuthorizationFilter.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+		if (tokenHeader == null) {
+			chain.doFilter(request, response);
+			return;
+		}
+
+		Map<String, String> payloadMap = getPayload(tokenHeader);
+		if (isValidCpcPlusOrg(payloadMap)) {
+			UsernamePasswordAuthenticationToken token =
+					new UsernamePasswordAuthenticationToken(payloadMap.get("id"), null , new ArrayList<>());
```
Comment (sfradkin) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/security/JwtAuthorizationFilter.java]: Please add some constants for map keys

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/security/JwtAuthorizationFilter.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
+package gov.cms.qpp.conversion.api.security;
+
+import com.google.common.collect.Sets;
```
Comment (FlexionSonarQube) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/security/JwtAuthorizationFilter.java]: MINOR Remove this unused import 'com.google.common.collect.Sets'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/security/JwtAuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
```
Comment (FlexionSonarQube) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/security/JwtAuthorizationFilter.java]: MINOR Remove this unused import 'java.util.Set'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by FlexionSonarQube** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**